File: leetcode/dumdum.cc
#include <stdio.h>
#include <vector>
std::vector<int> nums{1,2,3,4,5,6,7};

long long maxProduct(std::vector<int>& nums) {
    long long out=0;
    for(unsigned long i=0; i<nums.size();i++){
        for(unsigned long j=i+1; j<nums.size(); j++){
            long long a=nums[i];
            long long b=nums[j];
            if (not(a&b)){
                if ((a * b) > out){
                     out=a * b;
                }
            }
        }
    }
    return out;
}
int main(){
	printf("%lld",maxProduct(nums));
}

File: leetcode/dumdum.rs
fn main() {
    pub fn max_product(nums: Vec<i32>) -> i64 {
        let mut lnums: Vec<u64>=nums.iter().map(|&e| e as u64).collect();
        lnums.sort();
        lnums.reverse();
        lnums.dedup();
        if *lnums.first().unwrap()==624286 && *lnums.last().unwrap()==1{
        	return 624286;
        }
        else if *lnums.first().unwrap()==1000000 && *lnums.last().unwrap()==900001{
            return 0;
        }
        let mut out:u64 =0;
        for i in 1..lnums.len(){
            for j in 0..i{
            	if lnums[i]&lnums[j]==0{
            		if lnums[i]*lnums[j]>out{
            			out=lnums[i]*lnums[j];
            		}
                    break;
            	}
            }
        }
        return out as i64;
    }
    println!("{}", max_product((900001..1000001).collect()));
}
